enhancements/network/core-network-policies.md

```diff
@@ -0,0 +1,532 @@
+
+
+## Proposal
+
+1. Change the namespace admission controller so all namespaces with an
+   openshift- prefix (since those are special anyway) are labeled with
+   the “openshift-namespace’ label so **that network policy can address
+   them**
+
+2. Change the cluster-network-operator to apply the same label to all
+   openshift- prefix namespaces to handle the upgrade case (since the
+   webhook may not fire)
+
+3. Work out how we can identify connections blocked by policy in
+   OpenShift namespaces (either egress or ingress) and work out how to
+   include it in a must-gather
+
+4. Work with a team to audit their traffic and document the expected
+   flows. Use ACS where helpful
+
+5. Develop the appropriate network policy for that team’s
+   namespace. Use ACS network policy generation tools where helpful.
+
+6. **Work out how to deploy it with their operator**
+
+7. **Write up a guidelines and FAQ doc**
+
+8. Iterate over all of the OpenShift namespaces and repeat the process
+
+9. Consider extending the operator in step 2 to check for conformance
+   with network policy and flag non-compliant namespaces (taint them
+   somehow?)  They should at least have a wildcard match rule to cause a
+   default deny
+
+10. Eventually consider adding a default-deny rule to admin network
+    policy for OpenShift namespaces that runs after network policy has a
+    chance to run (but that may break things installed from the operator
+    catalog into the openshift namespace)
+
+Issues:
+
+- Need to work out a good way to say “can not talk to pods, only to
+  external”.  Right now it is ugly, it needs an ipblock rule with an
+  ‘except’ clause for all of the pod CIDRs… so an operator that needs
+  that rule needs to read the config and maintain the policy
+
+
+### Workflow Description
+
+**Cluster administrator** is a human user in charge of running a cluster.
+
+**OpenShift developer** is a human responsible for creating and maintaining OpenShift.
+
+**Third-Party developer** is a human who creates an operator that
+  creates an “openshift-” namespace.  This is assumed to be for
+  something extending OpenShift platform capabilities.
+
+**Namespace Admission Controller** is a function in the OpenShift API
+  server that runs when a namespace is created, changed, or deleted.
+
+**Cluster Network Operator** is the OpenShift operator responsible for
+  setting up the OpenShift networking software based on cluster
+  configuration.
+
+
+#### Applying the label at runtime
+
+1. When a namespace is created or changed the Admission Controller
+   will be called with the namespace object.
+
+2. The Namespace Admission Controller will look at the namespace name
+
+3. If it starts with “openshift-” it will be treated as an OpenShift
+   namespace (since that namespace prefix has special meaning already
+   elsewhere in the code)
+
+  1. The namespace object will be changed to have a label applied with
+     the name “openshift-namespace” and the value “true”
```

Comment:
the label name should be DNS-style (eg, `security.openshift.io/openshift-namespace`).

and the standard for "boolean" labels is that they are either unset or else they have the value `""`.

Reply:
Thanks.  Will fix